### ISSUE TITLE
Snowflake incremental load error on value lookup due to string literal

### ISSUE THREAD

**Issue by JoeNaso** [slingdata-io/sling-cli#85]
Incremental loads on Snowflake are failing due to string literal within SELECT query.  The double quotations wrapping the update_key reference are not compatible with Snowflake query engine in this case. 

For instance: 
`select max("updated_at") from my_table;` is invalid. Resulting in  `Error: invalid identifier '"updated_at"'`

That same query without the double quotes will work just fine
`select max(updated_at) from my_table;`

Here's a trace with table name redacted:
```
~ execution failed
--- task_run.go:615 runDbToDb ---
~ Could not get incremental value
--- task_func.go:213 getIncrementalValue ---
~ could not get max value for updated_at
--- database.go:1199 Query ---
~ Error with StreamRows
--- database.go:852 StreamRows ---
~ SQL Error for:
select max("updated_at") as max_val from "<REDACTED>"
--- database.go:890 StreamRowsContext ---
000904 (42000): SQL compilation error: error line 1 at position 11
invalid identifier '"updated_at"'
```

**Comment by flarco** [comment]
Hi, can you share your config? Also, what is the source connection type?

**Comment by JoeNaso** [comment]
Source conn is Postgres. 
Config snippets:

```
defaults:
  mode: incremental
  object: "{target_schema}.{stream_table}"
  # source_options:
  target_options:
    add_new_columns: true
    datetime_format: 2006-01-02 15:04:05.000000 -07


streams:
  public.<redacted>:
    primary_key:
      - id
    update_key: updated_at

**Comment by flarco** [comment]
what sling version are you using?

**Comment by JoeNaso** [comment]
My bad, meant to include that in the original post

[90m3:11PM[0m [32mINF[0m Downloading latest version (1.0.64)

**Comment by JoeNaso** [comment]
That table was likely created with an older version of sling. Does that affect behavior?

**Comment by flarco** [comment]
No it shouldn't. Just wondering.
Can you confirm you ran with 1.0.64?

**Comment by JoeNaso** [comment]
I just kicked off another job to grab fresh log - same deal. 

```
[2023-12-12 16:41:11] Updating Sling binary...
[90m4:41PM[0m [32mINF[0m Downloading latest version (1.0.64)
[90m4:41PM[0m [32mINF[0m Updated to 1.0.64
Version: 1.0.64
```
And further down the run: 
```invalid identifier '"updated_at"'```

I'm also seeing that it's unable to determine the target object name, which is unexpected 
```[90m2023-12-12 16:41:23[0m [33mDBG[0m Could not successfully format target object name. Blank values for: TARGET_SCHEMA```

**Comment by flarco** [comment]
OK, that may be the issue. For `{target_schema}` to work, you should make sure to specify the `schema` key in your snowflake connection (see https://docs.slingdata.io/connections/database-connections/snowflake). Or input the literal schema name instead of `{target_schema}`.

**Comment by JoeNaso** [comment]
Ok that makes sense, and we're doing that via the CLI. We're executing the binary via bash script, and setting the conn within that script, as well. We also test the connections and they work fine. See below:

```export SNOWFLAKE_DEST="type=snowflake \
    host=${SNOWFLAKE_ACCOUNT} \
    user=${SNOWFLAKE_USER} \
    password=${SNOWFLAKE_PASSWORD} \
    database=${SNOWFLAKE_DATABASE} \
    warehouse=${SNOWFLAKE_WAREHOUSE} \
    role=${SNOWFLAKE_ROLE} \
    schema=<hard-coded-value>"
```
Logs:
```
[90m4:41PM[0m [32mINF[0m connection `POSTGRES_SOURCE` has been set in /root/.sling/env.yaml. Please test with `sling conns test POSTGRES_SOURCE`
[90m4:41PM[0m [32mINF[0m connection `SNOWFLAKE_DEST` has been set in /root/.sling/env.yaml. Please test with `sling conns test SNOWFLAKE_DEST`
[2023-12-12 16:41:17] Testing Sling connections...
[90m4:41PM[0m [32mINF[0m success!
[90m4:41PM[0m [32mINF[0m success!
```
But Sling is later attempting to create a schema is already knows exists. Elsewhere in the logs I am still seeing the invalid identifier for the incremental value query
```
~ execution failed
--- task_run.go:91 func1 ---
~ Could not WriteToDb
--- task_run.go:641 runDbToDb ---
~ could not create temp table "<REDACTED_TEMP_TABLE>"
--- task_run_write.go:162 WriteToDb ---
~ Error checking & creating schema 
--- task_func.go:66 createTableIfNotExists ---
~ Error creating schema "<REDACTED>"
--- task_func.go:40 createSchemaIfNotExists ---
~ Could not execute SQL
--- database.go:1102 Exec ---
~ Error executing [tx: false] create schema "<REDACTED>"
--- database.go:1149 ExecContext ---
002002 (42710): SQL compilation error:
Object '<REDACTED>' already exists.
```

**Comment by flarco** [comment]
Interesting, can you try something? Can you ensure the schema name in `schema=<hard-coded-value>` is lowercased?

Looking at: https://github.com/slingdata-io/sling-cli/blob/a0f2ddc90404063927762a7d8fcd3e8fafccdb8d/core/sling/task_func.go#L29-L40 I believe there is a fix needed to make sure the letter casing is compared correctly (line 33 should be `schemaName = strings.ToLower(conn.GetProp("schema"))` so that line 39 is not triggered)

**Comment by flarco** [comment]
Should be fixed in `v1.0.65`.
